dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java

```diff
@@ -286,15 +275,19 @@ protected Identifier createNewIdentifier(Versionable versionable, Folder folder)
 	@Override
 	protected Identifier createNewIdentifier(Versionable versionable, Folder folder, String existingId) throws DotDataException {
 		User systemUser = APILocator.getUserAPI().getSystemUser();
-		String uuid=existingId;
 		Identifier identifier = new Identifier();
+
+		if (existingId !=  null) {
```

Comment:
you can replace it by
identifier.setId((existingId !=  null) ?existingId:UUIDGenerator.generateUuid());

Or you can even create a new method UUIDGenerator.generateUuid(existingId);